### PR TITLE
Decouple galaxy-app and galaxy-selenium.

### LINKED ISSUE #14187: galaxy-app package should depend on galaxy-selenium?
Since https://github.com/galaxyproject/galaxy/pull/14036 , files in `lib/galaxy/tours/` import from `galaxy.selenium.data` . So, since `tours` is included in galaxy-app, this should depend on galaxy-selenium?
Also, that may mean that we need to move selenium from dev dependency to main dependencies in `pyproject.toml` ?

### PR DESCRIPTION
**Pull request by jmchilton**
That was a lazy shortcut on my part and should have never happened. Fixes https://github.com/galaxyproject/galaxy/issues/14187. 

This resolves that and also another task on my todo list of making an easily pip installable, lightweight way of getting access to the tours validation script (introduced in https://github.com/galaxyproject/galaxy/pull/14003) at the same time by restructuring packages and cleaning up the dependency structure.

The navigation piece of galaxy-selenium has been split out of into its own package and tours now depend on that directly. In order to have that lightweight access to the tour validation code - tours have also been split out of app. This will also encourage that package to continue to be decoupled nicely from the larger galaxy-app package.

The new effective package structure should be...

core_packages plantuml

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: packages/tours/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+long_description_content_type = text/x-rst
+name = galaxy-selenium
+url = https://github.com/galaxyproject/galaxy
+version = 22.5.0.dev0
+
+[options]
+include_package_data = True
+install_requires =
+    galaxy-navigation
+    PyYAML
```
Comment (nsoranzo) [inline packages/tours/setup.cfg:36]:
```suggestion
    pydantic
    PyYAML
```

File: packages/navigation/setup.cfg [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+install_requires =
+    galaxy-util
+    PyYAML
+packages = find:
+python_requires = >=3.7
+
+[options.entry_points]
+console_scripts =
+        gx-dump-tour = galaxy.selenium.scripts.dump_tour:main
+
```
Comment (nsoranzo) [inline packages/navigation/setup.cfg]:
```suggestion
```

**Comment by nsoranzo** [comment]
Are you fine with https://github.com/galaxyproject/galaxy/pull/14150 , BTW? I'm not sure we should copy/paste selenium code, which is licensed under the Apache 2.0 license.

**Comment by nsoranzo** [comment]
Can you remove the `packages/app/galaxy/tours` symlink?

**Comment by jmchilton** [comment]
I'll redo it to restrict what we're using from Selenium here and to make it pretty clear the result isn't a copy and paste. Thanks for the licensing catch @nsoranzo.

File: lib/galaxy/selenium/smart_components.py [hunk 3]
```diff
@@ -7,7 +7,8 @@
 class SmartComponent:
     """Wrap a Component with driver aware methods.
 
-    Allows smarter selectors that know how to wait for themselves, test themselves,
+    Adapt's Galaxy's component locators more tightly to Selenium - inculding a Selenium
```
Comment (nsoranzo) [inline lib/galaxy/selenium/smart_components.py]:
```suggestion
    Adapts Galaxy's component locators more tightly to Selenium - including a Selenium
```

File: packages/tours/setup.cfg [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    Operating System :: POSIX
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Topic :: Software Development
+    Topic :: Software Development :: Code Generators
+    Topic :: Software Development :: Testing
+description = Galaxy Selenium interaction framework
```
Comment (nsoranzo) [inline packages/tours/setup.cfg]:
```suggestion
description = Galaxy tours backend framework
```

File: packages/tours/setup.cfg [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+keywords =
+    Galaxy
+license = AFL
+license_files =
+    LICENSE
+long_description = file: README.rst, HISTORY.rst
+long_description_content_type = text/x-rst
+name = galaxy-selenium
```
Comment (nsoranzo) [inline packages/tours/setup.cfg]:
```suggestion
name = galaxy-tours
```

File: packages/navigation/setup.cfg [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    Operating System :: POSIX
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Topic :: Software Development
+    Topic :: Software Development :: Code Generators
+    Topic :: Software Development :: Testing
+description = Galaxy Selenium interaction framework
```
Comment (nsoranzo) [inline packages/navigation/setup.cfg]:
```suggestion
description = Galaxy client DOM navigation framework
```

File: packages/tours/setup.cfg (hunk 4 above)
Comment (jmchilton) [reply packages/tours/setup.cfg]: I wonder if we should just auto-generate all these READMEs from setup.cfg fields 🤔.

**Pull request merged**